kn/fotos/entities.py

```diff
@@ -205,8 +249,30 @@ def get_parent(self):
             return None
         return by_path(self.path)
 
-    def set_title(self, title):
+    def set_title(self, title, save=True):
         self.title = title
+        if save:
+            self.save()
+
+    def update_visibility(self, visibility):
+        if self.visibility == visibility:
+            return
+
+        # First delete all old cached visibilities, in case something goes
+        # wrong during the update.
+        if self.path is None:
+            # root
+            query = {'path': {'$exists': True, '$ne': None}}
+        else:
+            query = {'path': {'$regex': re.compile(
+                                "^%s(/|$)" % re.escape(self.full_path))}}
+        fcol.update(query, {'$set': {'cachedVisibility': None}}, multi=True)
+
+        # And now save and recalculate cached visibilities recursively.
+        self.visibility = visibility
+        self.cached_visibility = None
+        self.save()
+        self.update_cached_visibility(self.get_parent())
```

Comment:
Nee, de parent is alleen nodig voor de `cachedVisibility`. Ik zou die op kunnen halen bij het updaten van de visibility, maar dat betekent een veelvoud in het aantal database queries, dus geef ik die als argument mee omdat die bij het recursief updaten al geladen is (logischerwijs).